types/errors/abci.go

```diff
@@ -1,11 +1,22 @@
 package errors
 
 import (
+	"math"
+
 	abci "github.com/cometbft/cometbft/v2/abci/types"
 
 	errorsmod "cosmossdk.io/errors"
 )
 
+// safeInt64FromUint64 converts uint64 to int64 with overflow checking.
+// If the value is too large to fit in int64, it returns math.MaxInt64.
+func safeInt64FromUint64(val uint64) int64 {
```

Comment:
Seems like we should just have this as a common function so we don't duplicate across the code